odinclient/src/main/kotlin/me/odinclient/features/impl/skyblock/ChocolateFactory.kt

```diff
@@ -94,4 +135,65 @@ object ChocolateFactory : Module(
         if (!cancelSound) return
         if (event.name == "random.eat") event.result = null // This should cancel the sound event
     }
+
+
+    private var currentDetectedEggs = arrayOfNulls<Egg>(3)
+
+    private enum class ChocolateEggs(
+        val texture: String, val type: String, val color: Color, val index: Int
+    ) {
+        Breakfast(BunnyEggTextures.breakfastEggTexture, "§6Breakfast Egg", Color.ORANGE, 0),
+        Lunch(BunnyEggTextures.lunchEggTexture, "§9Lunch Egg ", Color.BLUE, 1),
+        Dinner(BunnyEggTextures.dinnerEggTexture, "§aDinner Egg", Color.GREEN, 2),
+    }
+
+    data class Egg(val entity: EntityArmorStand, val renderName: String, val color: Color, var isFound: Boolean = false)
+
+    private fun getEggType(entity: EntityArmorStand): ChocolateEggs? {
+        val texture = getSkullValue(entity)
+        return ChocolateEggs.entries.find { it.texture == texture }
+    }
+
+    fun scanForEggs() {
```

Comment:
You should filter the list instead of checking it while iterating over it 
Also switch to foreach should look nicer